### PR TITLE
Add TRL tuning instructions for gemma 2

### PR DESCRIPTION
**Pull request by vwxyzjn**
DS3: https://wandb.ai/huggingface/trl/runs/s4um6ghd?nw=nwusercostahuang
LORA: https://wandb.ai/huggingface/huggingface/runs/v4fd7y8g?nw=nwusercostahuang

### CODE REVIEW

File: gemma2.md [hunk 1]
```diff
@@ -213,6 +213,84 @@ For more details on using the models with transformers, please check [the model
...
+## Fine-tuning with 🤗 TRL
+
+Training LLMs can be technically and computationally challenging. In this section, we’ll look at the tools available in the Hugging Face ecosystem to efficiently train Gemma on consumer-size GPUs
+
+An example command to fine-tune Gemma on OpenAssistant’s [chat dataset](https://huggingface.co/datasets/OpenAssistant/oasst_top1_2023-08-25) can be found below. We use 4-bit quantization and [QLoRA](https://arxiv.org/abs/2305.14314) to conserve memory to target all the attention blocks' linear layers. Note that, unlike dense transformers, one should not target the MLP layers as they are sparse and don’t interact well with PEFT.
+
+First, install the nightly version of 🤗 TRL and clone the repo to access the [training script](https://github.com/huggingface/trl/blob/main/examples/scripts/sft.py):
+
+```jsx
+pip install "transformers==4.42.0" --upgrade
```
Comment (kashif) [inline gemma2.md:227]:
```suggestion
pip install "transformers==4.42.1"
```

File: zh/gemma2.md [hunk 2]
```diff
@@ -216,6 +216,80 @@ pipeline = pipeline(
...
+## 使用 🤗 TRL 进行微调
+
+训练大型语言模型在技术和计算上都具有挑战性。在本节中,我们将了解 Hugging Face 生态系统中可用的工具,以便在消费级 GPU 上高效训练 Gemma。
+
+下面是在 OpenAssistant 的[聊天数据集](https://huggingface.co/datasets/OpenAssistant/oasst_top1_2023-08-25)上微调 Gemma 的示例命令。我们使用 4 位量化和 [QLoRA](https://arxiv.org/abs/2305.14314) 来节省内存,以针对所有注意力块的线性层。请注意,与密集变换器不同,不应针对 MLP 层,因为它们是稀疏的,与 PEFT 不太兼容。
+
+首先,安装 🤗 TRL 的每日版本并克隆仓库以访问[训练脚本](https://github.com/huggingface/trl/blob/main/examples/scripts/sft.py):
+
+```jsx
+pip install "transformers==4.42.0" --upgrade
```
Comment (kashif) [inline zh/gemma2.md:229]:
```suggestion
pip install "transformers==4.42.1"
```

File: gemma2.md [hunk 3]
```diff
@@ -213,6 +213,84 @@ For more details on using the models with transformers, please check [the model
...
+pip install --upgrade bitsandbytes
+pip install --ugprade peft
+pip install git+https://github.com/huggingface/trl
+git clone https://github.com/huggingface/trl
+cd trl
+```
+
+Then you can run the script:
+
+```jsx
```
Comment (philschmid) [inline gemma2.md:237]:
```suggestion
```bash
```

**Review by philschmid — COMMENTED** [review]
some minor comments

File: gemma2.md [hunk 4]
```diff
@@ -213,6 +213,84 @@ For more details on using the models with transformers, please check [the model
...
+## Fine-tuning with 🤗 TRL
+
+Training LLMs can be technically and computationally challenging. In this section, we’ll look at the tools available in the Hugging Face ecosystem to efficiently train Gemma on consumer-size GPUs
+
+An example command to fine-tune Gemma on OpenAssistant’s [chat dataset](https://huggingface.co/datasets/OpenAssistant/oasst_top1_2023-08-25) can be found below. We use 4-bit quantization and [QLoRA](https://arxiv.org/abs/2305.14314) to conserve memory to target all the attention blocks' linear layers. Note that, unlike dense transformers, one should not target the MLP layers as they are sparse and don’t interact well with PEFT.
+
+First, install the nightly version of 🤗 TRL and clone the repo to access the [training script](https://github.com/huggingface/trl/blob/main/examples/scripts/sft.py):
+
+```jsx
+pip install "transformers==4.42.1"
```
Comment (kashif) [inline gemma2.md:227]:
```suggestion
pip install "transformers==4.42.2"
```

File: zh/gemma2.md [hunk 5]
```diff
@@ -216,6 +216,80 @@ pipeline = pipeline(
...
+## 使用 🤗 TRL 进行微调
+
+训练大型语言模型在技术和计算上都具有挑战性。在本节中,我们将了解 Hugging Face 生态系统中可用的工具,以便在消费级 GPU 上高效训练 Gemma。
+
+下面是在 OpenAssistant 的[聊天数据集](https://huggingface.co/datasets/OpenAssistant/oasst_top1_2023-08-25)上微调 Gemma 的示例命令。我们使用 4 位量化和 [QLoRA](https://arxiv.org/abs/2305.14314) 来节省内存,以针对所有注意力块的线性层。请注意,与密集变换器不同,不应针对 MLP 层,因为它们是稀疏的,与 PEFT 不太兼容。
+
+首先,安装 🤗 TRL 的每日版本并克隆仓库以访问[训练脚本](https://github.com/huggingface/trl/blob/main/examples/scripts/sft.py):
+
+```jsx
+pip install "transformers==4.42.1"
```
Comment (kashif) [inline zh/gemma2.md:229]:
```suggestion
pip install "transformers==4.42.2"
```

File: zh/gemma2.md [hunk 6]
```diff
@@ -216,6 +216,80 @@ pipeline = pipeline(
...
+pip install --upgrade bitsandbytes
+pip install --ugprade peft
+pip install git+https://github.com/huggingface/trl
+git clone https://github.com/huggingface/trl
+cd trl
+```
+
+然后你可以运行该脚本:
+
+```jsx
```
Comment (kashif) [inline zh/gemma2.md:239]:
```suggestion
```bash
```

File: zh/gemma2.md [hunk 7]
```diff
@@ -216,6 +216,80 @@ pipeline = pipeline(
...
+	--output_dir models/gemma2
+```
+
+<p align="center">
+  <img src="https://huggingface.co/datasets/trl-internal-testing/example-images/resolve/main/blog/gemma2/lora.png?download=true" alt="alt_text" title="image_tooltip" />
+</p>
+
+如果你有更多的 GPU 可用,可以使用 DeepSpeed 和 ZeRO Stage 3 进行训练:
+
+```jsx
```
Comment (kashif) [inline zh/gemma2.md:269]:
```suggestion
```bash
```

File: gemma2.md [hunk 8]
```diff
@@ -213,6 +213,84 @@ For more details on using the models with transformers, please check [the model
...
+	--dataset_name OpenAssistant/oasst_top1_2023-08-25 \
+	--dataset_text_field="text" \
+	--per_device_train_batch_size 1 \
+	--per_device_eval_batch_size 1 \
+	--gradient_accumulation_steps 4 \
+	--learning_rate 2e-5 \
+	--report_to wandb \
+	--bf16 \
+	--max_seq_length 1024 \
+	--attn_implementation flash_attention_2 \
```
Comment (kashif) [inline gemma2.md:282]:
```suggestion
	# --attn_implementation flash_attention_2 \ # gemma 2 supports FA2 only for inference
```

File: gemma2.md [hunk 9]
```diff
@@ -213,6 +213,84 @@ For more details on using the models with transformers, please check [the model
...
+	--gradient_accumulation_steps 4 \
+	--learning_rate 2e-4 \
+	--report_to wandb \
+	--bf16 \
+	--max_seq_length 1024 \
+	--lora_r 16 --lora_alpha 32 \
+	--lora_target_modules q_proj k_proj v_proj o_proj \
+	--load_in_4bit \
+    --use_peft \
+	--attn_implementation flash_attention_2 \
```
Comment (kashif) [inline gemma2.md:255]:
```suggestion
	# --attn_implementation flash_attention_2 \ # gemma 2 supports FA2 only for inference
```

File: zh/gemma2.md [hunk 10]
```diff
@@ -216,6 +216,80 @@ pipeline = pipeline(
...
+	--gradient_accumulation_steps 4 \
+	--learning_rate 2e-4 \
+	--report_to wandb \
+	--bf16 \
+	--max_seq_length 1024 \
+	--lora_r 16 --lora_alpha 32 \
+	--lora_target_modules q_proj k_proj v_proj o_proj \
+	--load_in_4bit \
+    --use_peft \
+	--attn_implementation flash_attention_2 \
```
Comment (kashif) [inline zh/gemma2.md:257]:
```suggestion
	# --attn_implementation flash_attention_2 \
```

**Review by kashif — APPROVED** [review]

**Pull request merged**